types/src/diem_id_identifier.rs

```diff
@@ -0,0 +1,289 @@
+// Copyright (c) The Diem Core Contributors
+// SPDX-License-Identifier: Apache-2.0
+
+use serde::{de, Serialize};
+use std::{
+    fmt::{Display, Formatter},
+    str::FromStr,
+};
+
+/// This file implements length and character set limited string types needed
+/// for DiemID as defined by DIP-10: https://github.com/diem/dip/blob/main/dips/dip-10.md
```

Comment:
```suggestion
//! This file implements length and character set limited string types needed
//! for DiemID as defined by DIP-10: https://github.com/diem/dip/blob/main/dips/dip-10.md
```

And maybe put this above the `use` statements.